src/ray_agents/adapters/langchain/agents.py

```diff
@@ -0,0 +1,293 @@
+"""LangChain agent adapter for Ray distributed tool execution."""
+
+import functools
+import logging
+from collections.abc import Callable
+from typing import Any
+
+import ray
+
+from ray_agents.adapters.abc import AgentAdapter
+
+logger = logging.getLogger(__name__)
+
+
+class LangChainAdapter(AgentAdapter):
+    """
+    Adapter for LangChain agents with Ray distributed tool execution.
+
+    This adapter wraps LangChain's native agent execution (ReAct pattern, tool
+    calling) while executing tools as Ray tasks. LangChain maintains full control
+    over the agent flow; Ray provides distributed execution.
+
+    When LangChain calls a tool, it's actually executing:
+        result = ray.get(tool.remote(**args))
+
+    This means tools automatically run on cluster nodes with appropriate resources.
+
+    Example:
+        >>> from ray_agents.adapters.langchain import LangChainAdapter
+        >>> from langchain_openai import ChatOpenAI
+        >>> import ray
+        >>>
+        >>> # Define tools with resource requirements
+        >>> @ray.remote(num_gpus=1)
+        >>> def tool_a(param: str):
+        ...     '''First tool with specific resource needs'''
+        ...     return {"output": "result_a"}
+        >>>
+        >>> @ray.remote(num_cpus=8, memory=4 * 1024**3)
+        >>> def tool_b(param: dict):
+        ...     '''Second tool with different resource needs'''
+        ...     return {"output": "result_b"}
+        >>>
+        >>> # Create adapter with any LangChain LLM
+        >>> llm = ChatOpenAI(model="gpt-4o-mini", temperature=0.7)
+        >>> adapter = LangChainAdapter(llm=llm)
+        >>>
+        >>> # Use with AgentSession for stateful conversations
+        >>> from ray_agents import AgentSession
+        >>> session = AgentSession.remote("user_id", adapter=adapter)
+        >>> result = ray.get(session.run.remote(
+        ...     "Use the available tools to help me",
+        ...     tools=[tool_a, tool_b]
```

Comment:
I think we should clean these up and put them in docs? Seems non-standard to me to have such docstrings in code. Let me know if you have examples from elsewhere